Tee'sApp/Controllers/ViewController.swift

```diff
@@ -0,0 +1,186 @@
+        tableView.dataSource = self
+        viewModel.delegate = self
+
+        //delegating search field so a keyboard push registers as the 'return; key
+        searchField.delegate = self
+        //networking data(:
+        viewModel.fetchData()
+        apiActivityIndicator.startAnimating()
+    }
+    
+    
+    @IBAction func searchButtonPressed(_ sender: UIButton) {
+        searchField.endEditing(true)
+        userName = searchField.text!
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let landingVC = segue.destination as? ViewControllerB else{
+            return
+        }
+        
+        //data we plan to pass to our landing ViewController
+        landingVC.info = userName
+        landingVC.userData = userData
+        landingVC.userIndex = userIndex
+        landingVC.user = user
+    }
+}
+
+// MARK: - UITableViewDelegate
+extension ViewController: UITableViewDelegate{
+    // Returns the number of rows for the table.
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return userData.count
+    }
+    
+    // Provides a cell object for each row.
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        // Fetch a cell of the appropriate type.
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cellone", for: indexPath)
+        
+        
+            let index = indexPath.row % data.count
+                    
+        let title = userData[index].username
+            
+            // configures cell’s contents.
+            cell.textLabel!.text = title
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        if let title = cell?.textLabel?.text {
+            print(title)
+            userName = userData[indexPath.row].name
+            userIndex = indexPath.row
+            user = userData[indexPath.row]
+            performSegue(withIdentifier: "first", sender: nil)
+        }
+            
+        }
+    
+    // MARK: - UIAlertController
+    func showSimpleAlert(description: String) {
+        // UIAlert that gives uer option to retry API call or to simple cancel
+        let alert = UIAlertController(title: "Error Pulling Data", message: description,   preferredStyle: UIAlertController.Style.alert)
+        
+        alert.addAction(UIAlertAction(title: "Retry", style: UIAlertAction.Style.default, handler: { (action)-> Void in
+            self.viewModel.fetchData()
+        }))
+
+        //Cancel Action
+        alert.addAction(UIAlertAction(title: "Cancel", style: UIAlertAction.Style.default, handler: { _ in
+        }))
+
+        self.present(alert, animated: true, completion: nil)
+    }
+        
+    
+}
+
+// MARK: - UITextFieldDelegate
+extension ViewController: UITextFieldDelegate{
```

Comment:
Breaking out protocol conformance into extensions is a very nice way to keep your VC clean!